src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -1521,6 +1521,48 @@ static void log_cb (const rd_kafka_t *rk, int level,
         CallState_resume(cs);
 }
 
+static void oauth_cb (rd_kafka_t *rk, const char *oauthbearer_config, void *opaque) {
+	Handle *h = opaque;
+	PyObject *eo = NULL, *result = NULL;
+	CallState *cs = NULL;
+	const char *token = NULL;
+	long long expiry = 0;
+	char err_msg[2048];
+	rd_kafka_resp_err_t err_code = RD_KAFKA_RESP_ERR_NO_ERROR;
+
+	cs = CallState_get(h);
+
+	eo = Py_BuildValue("s", oauthbearer_config);
+	result = PyObject_CallFunctionObjArgs(h->oauth, eo, NULL);
+	Py_DECREF(eo);
+
+	if (!result) {
+	    goto err;
+	} else {
+	    if (!PyArg_ParseTuple(result, "sl", &token, &expiry)) {
+	        Py_DECREF(result);
+	        PyErr_Format(PyExc_TypeError,
+	                     "expect returned value from oauth "
+	                     "to be (token_str, expiry_int_ms) tuple");
+	        goto err;
+	    }
+	}
+	err_code = rd_kafka_oauthbearer_set_token(h->rk, token, expiry, "", NULL, 0, err_msg, sizeof(err_msg));
+	Py_DECREF(result);
+	if (err_code) {
+	    PyErr_Format(PyExc_ValueError,
+	                 "invalid token: %s", err_msg);
```

Comment:
I think you should remove "invalid token: " and just keep the "%s" as the error message from librdkafka should be complete as-is.